bfvmm/include/vmm.h

```diff
@@ -81,6 +83,9 @@
 
 #ifdef BF_INTEL_X64
 using namespace bfvmm::intel_x64;
+
+#define handler_delegate(function) handler_delegate_t::create<function>()
```

Comment:
this is going to cause problems because it will collide with the existing code. This is also not helpful because it prevents the use of classes in the delegate. Just use the long form. Its long, but at least it is explicit as to what you are doing.

Reply:
Maybe just use a different name for the macro, like ```handler()```? What does it collide with?

I find ```handler_delegate_t::create<function>()``` not simple, and just ugly. Also, I disagree that this prevents the use of classes in the delegate, you can always use the non-macro syntax if you want it.

Reply:
handler_delegate is contained in handler_delegate_t (same with handler). Downstream files could end up with collisions. With that said, I am fine with trying to find a meta programming approach to reducing their verbosity as the class version of this can get along quickly with no great way to line break. 

Also... I completely disagree with the function version of this. I think it is very elegant. It reads just like a std::unique_ptr<>. Its the class version that sucks. @connojd and I spent some time to try and make the class version a little less verbose without success but I am open to suggestions that don't leverage macros where possible. We have a bunch of these in Boxy as well that I want to remove as well.

Reply:
Connor just demo'ed me a far superior solution to the macro that gives pretty much the same syntax. I'll remove the macro

Reply:
Yeah. IMO, worst case, we can address this issue using the malloc based delegate which has a really nice and clean interface. We would just have to ensure we know what the malloc size is so that we can create an object_allocator for that allocation size. Since we don't register delegate at runtime, this approach would work too. But yeah, if Connor can get his C++17 type deduction trick to work in the class case, that is the way to do it.